sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -82,106 +83,64 @@ def check_classification_toy(loss):
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(min_samples_leaf=-1.0).fit, X, y)
-
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(min_weight_fraction_leaf=-1.).fit,
-                  X, y)
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(min_weight_fraction_leaf=0.6).fit,
-                  X, y)
-
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(subsample=0.0).fit, X, y)
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(subsample=1.1).fit, X, y)
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(subsample=-0.1).fit, X, y)
-
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(max_depth=-0.1).fit, X, y)
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(max_depth=0).fit, X, y)
-
-    assert_raises(ValueError,
-                  GradientBoostingClassifier(init={}).fit, X, y)
-
-    # test fit before feature importance
-    assert_raises(ValueError,
-                  lambda: GradientBoostingClassifier().feature_importances_)
-
-    # deviance requires ``n_classes >= 2``.
-    assert_raises(ValueError,
-                  lambda X, y: GradientBoostingClassifier(
-                      loss='deviance').fit(X, y),
-                  X, [0, 0, 0, 0])
-
-
-def test_regressor_parameter_checks():
+def _check_parameter_error(Estimator, params, TypeError, err_msg=None):
+    with pytest.raises(TypeError, match=err_msg):
+        Estimator(**params).fit(X, y)
+
+
+@pytest.mark.parametrize(
+    "params",
+    [{"n_estimators": 0}, {"n_estimators": -1},
+     {"learning_rate": 0}, {"learning_rate": -1.0},
+     {"loss": "foobar"},
+     {"min_samples_split": 0.0}, {"min_samples_split": -1.0},
+     {"min_samples_split": 1.1},
+     {"min_samples_leaf": 0}, {"min_samples_leaf": -1.0},
+     {"min_weight_fraction_leaf": -1.0}, {"min_weight_fraction_leaf": 0.6},
+     {"subsample": 0.0}, {"subsample": 1.1}, {"subsample": -0.1},
+     {"max_depth": -0.1}, {"max_depth": 0},
+     {"init": {}}]
+)
+def test_classifier_parameter_checks(params):
+    # Check input parameter validation for GradientBoostingClassifier
+    _check_parameter_error(GradientBoostingClassifier, params, ValueError)
+
+
+@pytest.mark.parametrize(
+    "params, err_msg",
+    [({"loss": "huber", "alpha": 1.2}, r"alpha must be in \(0.0, 1.0\)"),
+     ({"loss": "quantile", "alpha": 1.2}, r"alpha must be in \(0.0, 1.0\)"),
+     ({"max_features": "invalid"}, "Invalid value for max_features:"),
+     ({"max_features": 0}, r"max_features must be in \(0, n_features\]"),
+     ({"max_features": len(X[0]) + 1}, "max_features must be in"),
+     ({"max_features": -0.1}, r"max_features must be in \(0, n_features\]"),
+     ({"n_iter_no_change": "invalid"}, "n_iter_no_change should either be")]
+)
+def test_regressor_parameter_checks(params, err_msg):
```

Comment:
apart from the loss checks it seems that all of these should also apply to the classifier? Should we parametrize the check above with the regressor for more coverage, and add to it all the checks that aren't about the loss?